### PR TITLE
Improvement/search results

### PR DESCRIPTION
**Pull request by lucieleonard**
This PR improves:
- Search result pages (text, variables)
- CSS
- Pages layout

### CODE REVIEW

File: sphinx_scality/static/css/current.css_t [hunk 1]
```diff
@@ -265,3 +265,8 @@ div.footer div.rightwrapper {
 div.footer div.rightwrapper img.social {
     margin: auto 15px;
 }
+
+.term {
+    color: #F37533;
```
Comment (gdemonet) [inline sphinx_scality/static/css/current.css_t]: Not sure exactly what a `.term` is, but:

- It probably doesn't belong among footer styles
- Colours shouldn't be hardcoded if possible, unless it doesn't make sense to change them when using a different colour palette (you can get inspiration from other uses of `{{ theme_<something>color }}` and the corresponding definition in `theme.conf`)

File: sphinx_scality/static/js/searchtools.js [hunk 2]
```diff
@@ -0,0 +1,508 @@
+/*
+ * searchtools.js
+ * ~~~~~~~~~~~~~~~~
+ *
+ * Sphinx JavaScript utilities for the full-text search.
```
Comment (gdemonet) [inline sphinx_scality/static/js/searchtools.js]: Let's split the addition of this `searchtools.js` file from the customization (in terms of Git commits), it will help with identifying what changes were introduced. Another approach could have been to import this JS module from another one we own and only override the methods we need, but that may be harder to implement.

File: sphinx_scality/static/js/searchtools.js [hunk 3]
```diff
@@ -0,0 +1,508 @@
...
+      }
+      // search finished, update title and status message
+      else {
+        Search.stopPulse();
+        Search.title.text(_('Results'));
+        if (!resultCount)
+          Search.status.html(_('Your search did not match any documents. Try browsing the \
+          <a href="https://support.scality.com/hc/en-us">Knowledge Base</a> or contact Scality Support.'));
+        else
+            Search.status.html(_('Found %s page(s) matching the term(s) "<span class="term">%t</span>". Still not what you are\
```
Comment (gdemonet) [inline sphinx_scality/static/js/searchtools.js]: IIRC, those are the lines you changed, and they are slightly too long (we prefer to maintain a maximum of 80 columns, for easy reading on split/vertical screens).

Also, and it might have been like this before, we could improve the syntax used for clarity (assumes you defined a constant somewhere called `KB_LINK`, could prove useful since you use it two lines above):

```javascript
Search.status.html(_(`
  Found ${resultCount} page(s) matching the term(s) "<span class="term">${searchterms}</span>".
  Still not what you're looking for? Try browsing the <a href="${KB_LINK}">Knowledge Base</a>
  or contact Scality Support.
`));
```

File: sphinx_scality/layout.html [hunk 4]
```diff
@@ -114,17 +115,22 @@ <h2><a href="{{ pathto(pagename) }}">{{ title }}</a></h2>
   {%- endblock %} {# sidebarleft #}
 
   {%- block content %}
-  <div class="content">
+    <div class="content inside">
```
Comment (gdemonet) [inline sphinx_scality/layout.html]: There are some indentation issues here and below.

File: sphinx_scality/layout.html [hunk 5]
```diff
@@ -114,17 +115,22 @@ <h2><a href="{{ pathto(pagename) }}">{{ title }}</a></h2>
...
       {% block body %} {% endblock %}
     </div>
   </div>
+        </div>
   {%- endblock %}
 
-  {%- block right_sidebar %}
-    <div class="sidebar right">
+  {%- block right_sidebar %}  
+
```
Comment (gdemonet) [inline sphinx_scality/layout.html]: Why the newline?

File: sphinx_scality/static/css/current.css_t [hunk 6]
```diff
@@ -77,7 +77,7 @@ div.sidebar {
     margin: 0;
     padding: 0 20px;
     width: auto;
-    overflow: auto;
+    overflow: visible;
```
Comment (gdemonet) [inline sphinx_scality/static/css/current.css_t]: Overflow of a sidebar should not be visible, otherwise you'll get weird results if there's any and the sidebar is fixed (content is forever lost out of the screen).

File: sphinx_scality/static/css/current.css_t [hunk 7]
```diff
@@ -269,4 +275,19 @@ div.footer div.rightwrapper img.social {
...
+  width: 300px;
+  box-sizing: border-box;
+  border: 1px solid #ccc;
+  border-radius: 20px;
+  background-image: url(../img/search-icon.png);
+  background-size: 35px 35px;
+  background-position: right;
+  background-repeat: no-repeat;
+  padding: 10px 20px 12px 30px;
+
```
Comment (gdemonet) [inline sphinx_scality/static/css/current.css_t]: Unneeded newline

File: sphinx_scality/static/css/current.css_t [hunk 8]
```diff
@@ -269,4 +275,19 @@ div.footer div.rightwrapper img.social {
...
+}
+
+/* search box */
+
+input[type=text], select {
+  width: 300px;
+  box-sizing: border-box;
+  border: 1px solid #ccc;
+  border-radius: 20px;
+  background-image: url(../img/search-icon.png);
```
Comment (gdemonet) [inline sphinx_scality/static/css/current.css_t]: Don't we need to use quotes with `url(...)`?

File: sphinx_scality/static/css/current.css_t [hunk 9]
```diff
@@ -269,4 +275,19 @@ div.footer div.rightwrapper img.social {
 .term {
     color: #F37533;
     font-weight: bold;
+}
+
+/* search box */
+
+input[type=text], select {
```
Comment (gdemonet) [inline sphinx_scality/static/css/current.css_t]: We need a more precise selector here, and I'm quite surprised by the `select` here, is it needed?

File: sphinx_scality/static/css/theme.css [hunk 10]
```diff
@@ -3,3 +3,122 @@
 
 @import url("./fonts.css");
 @import url("./current.css");
+
```
Comment (gdemonet) [inline sphinx_scality/static/css/theme.css]: Those styles should not be in `theme.css`, which is just an entrypoint for the others. If you want to split header from footer and the likes, create new files to hold those styles, and keep things neat. Otherwise, we'll never know where the styles for the header are since they are disseminated all over the place.

File: sphinx_scality/static/css/theme.css [hunk 11]
```diff
@@ -3,3 +3,122 @@
...
+
+/****** header******/
```
Comment (gdemonet) [inline sphinx_scality/static/css/theme.css]: Comments syntax is `/* <message> */`, let's try to be consistent and not add a random number of stars. Welcome to the rigid world of development ;)

File: sphinx_scality/static/css/theme.css [hunk 12]
```diff
@@ -3,3 +3,122 @@
...
+
+
+
+/**** Header logo arrow search *****/
+
+img.baselogo {
+    width: 125px;
+}
+div.baselogo > a {
+
```
Comment (gdemonet) [inline sphinx_scality/static/css/theme.css]: Let's clean spacing please: each CSS rule should be surrounded with empty lines (1 above, 1 below), and a rule shouldn't contain empty lines.

File: sphinx_scality/static/css/theme.css [hunk 13]
```diff
@@ -3,3 +3,122 @@
...
+div.logo img {
+    width: 133px;
+}
+
+
+
+/*****content******/
+
+div.content.inside {
+    margin: 0% 5% 0 10% !important;
```
Comment (gdemonet) [inline sphinx_scality/static/css/theme.css]: Why do we need `!important` here? It really is a [bad practice](https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity#The_!important_exception), and since we own the HTML, we just need to bump the specificity or use another selector that doesn't collide with other rules.

File: sphinx_scality/static/css/theme.css [hunk 14]
```diff
@@ -3,3 +3,122 @@
...
+
+/***** side bar *****/
+
+div.sidebar {
+   
+    width: 74%;
+}
+
+div.sidebar {
+    margin: 0 0% 0 42% !important;
```
Comment (gdemonet) [inline sphinx_scality/static/css/theme.css]: ditto

File: sphinx_scality/static/css/current.css_t [hunk 15]
```diff
@@ -290,4 +290,11 @@ input[type=text], select {
   background-repeat: no-repeat;
   padding: 10px 20px 12px 30px;
 
+}
+
+/* search results */
+
+.docnames {
+    font-size: 10px;
+    font-color: #a9a9a9;
```
Comment (gdemonet) [inline sphinx_scality/static/css/current.css_t]: Ditto, warrants a theme option IMO

File: sphinx_scality/static/js/searchtools.js [hunk 16]
```diff
@@ -26,7 +26,7 @@ if (!Scorer) {
     // query matches the full name of an object
     objNameMatch: 11,
     // or matches in the last dotted part of the object name
-    objPartialMatch: 6,
+    objPartialMatch: 15,
```
Comment (gdemonet) [inline sphinx_scality/static/js/searchtools.js]: I don't understand this change. What does it do? Would be worth explaining in the commit message.

File: sphinx_scality/static/css/current.css_t [hunk 17]
```diff
@@ -233,6 +233,19 @@ div.body {
     margin: 40px 20px;
 }
 
+/* LINKS */
+
+a:visited {
+    color: #ff9b3b;
```
Comment (gdemonet) [inline sphinx_scality/static/css/current.css_t]: Ditto, theme options FTW

File: sphinx_scality/static/css/current.css_t [hunk 18]
```diff
@@ -233,6 +233,19 @@ div.body {
     margin: 40px 20px;
 }
 
+/* LINKS */
+
+a:visited {
```
Comment (gdemonet) [inline sphinx_scality/static/css/current.css_t]: If those rules really need to apply to all links in the HTML, then I guess some others could be cleaned-up (at least I remember adding a few `text-decoration: none` in other places).

File: sphinx_scality/static/css/current.css_t [hunk 19]
```diff
@@ -297,4 +310,109 @@ input[type=text], select {
 .docnames {
     font-size: 10px;
     font-color: #a9a9a9;
-}
\ No newline at end of file
+}
+
+/* TITLES */
+
+div.body h1 { margin-top: 0; font-size: 200%; }
```
Comment (gdemonet) [inline sphinx_scality/static/css/current.css_t]: Can we try to use `em`s instead of percents for font sizes?

File: sphinx_scality/static/css/current.css_t [hunk 20]
```diff
@@ -297,4 +310,109 @@ input[type=text], select {
...
+    color: #3F5952;
+    text-decoration: none;
+}
+
+/* CODE */
+
+pre
+  {   font-size: 90%;
+      font-family: Courier, "Courier New", monospace ; 
+/*      margin-left: 18pt;*/
```
Comment (gdemonet) [inline sphinx_scality/static/css/current.css_t]: If unused, can be removed.

File: sphinx_scality/static/css/current.css_t [hunk 21]
```diff
@@ -297,4 +310,109 @@ input[type=text], select {
...
+      margin-top: 0pt;
+      color: '#000000';
```
Comment (gdemonet) [inline sphinx_scality/static/css/current.css_t]: Theme options!

File: sphinx_scality/static/css/current.css_t [hunk 22]
```diff
@@ -297,4 +310,109 @@ input[type=text], select {
...
+      background-color: '#F2F2F2';
+      border-bottom: none;
```
Comment (gdemonet) [inline sphinx_scality/static/css/current.css_t]: `border: none` should be sufficient for those four lines.

File: sphinx_scality/static/css/current.css_t [hunk 23]
```diff
@@ -297,4 +310,109 @@ input[type=text], select {
...
+      border-bottom: none;      
+      border-top: none;
+      border-left: none;
+      border-right: none;
+  }
+
+/* TABLES */
+
+table.docutils th {
+  background-color: #eaeaea;
```
Comment (gdemonet) [inline sphinx_scality/static/css/current.css_t]: Ditto

File: sphinx_scality/static/css/current.css_t [hunk 24]
```diff
@@ -297,4 +310,109 @@ input[type=text], select {
...
+}
+
+table caption {
+    font-weight: bold;
+    text-align: left;
+    margin-top: 20px;
+}
+
+table.docutils td, table.docutils th {
+    border-top: 1px solid #aaa;
```
Comment (gdemonet) [inline sphinx_scality/static/css/current.css_t]: Ditto

File: sphinx_scality/static/css/current.css_t [hunk 25]
```diff
@@ -297,4 +310,109 @@ input[type=text], select {
...
+}
+
+/* IMAGES */
+
+img
+  {
```
Comment (gdemonet) [inline sphinx_scality/static/css/current.css_t]: Let's open this bracket on the same line as the selector.

File: sphinx_scality/static/css/current.css_t [hunk 26]
```diff
@@ -297,4 +310,109 @@ input[type=text], select {
...
+    margin-top: 20px;
+}
+
+table.docutils td, table.docutils th {
+    border-top: 1px solid #aaa;
+}
+
+/* IMAGES */
+
+img
```
Comment (gdemonet) [inline sphinx_scality/static/css/current.css_t]: Not sure we want global styles for images, feels quite risky.

File: sphinx_scality/static/css/current.css_t [hunk 27]
```diff
@@ -297,4 +310,109 @@ input[type=text], select {
...
+  {
+    align: center;
+    padding-left: 10px;
+    padding-right: 10px;
+    padding-top: 5px;
+    padding-bottom: 5px;
+  }
+
+
+img.FiftyPercent
```
Comment (gdemonet) [inline sphinx_scality/static/css/current.css_t]: What are those class names? Did we introduce them? Because they are quite non-standard.

File: sphinx_scality/static/css/current.css_t [hunk 28]
```diff
@@ -297,4 +310,109 @@ input[type=text], select {
...
+  {
+    max-width: 50%;
+  }
+
+
+img.OneHundredPercent
+  {
+    max-width: 100%;
+/*    padding-left: 10px;
```
Comment (gdemonet) [inline sphinx_scality/static/css/current.css_t]: Unused = removed

File: sphinx_scality/static/css/current.css_t [hunk 29]
```diff
@@ -297,4 +310,109 @@ input[type=text], select {
...
+    padding-right: 10px;
+    padding-top: 5px;
+    padding-bottom: 5px;*/
+  }
+
+img.SeventyFivePercent
+  {
+    align: center;
+    max-width: 75%;
+  /*      padding-left: 10px;
```
Comment (gdemonet) [inline sphinx_scality/static/css/current.css_t]: Ditto.

File: example/conf.py [hunk 30]
```diff
@@ -106,6 +106,7 @@
 
 html_show_sphinx = False
 html_show_source = False
+html_add_permalinks = '#'
```
Comment (gdemonet) [inline example/conf.py]: How is this related to `Enhanced titles`? I don't even know what it does, maybe an explanation in the commit message or in a comment could help.

File: sphinx_scality/footer.html [hunk 31]
```diff
@@ -10,16 +10,7 @@
       <a href="#">Privacy Policy</a> |
       <a href="#">Training</a>
     </div>
-    <div class="copyright">
```
Comment (gdemonet) [inline sphinx_scality/footer.html]: No more copyrights? Why not? And what's the link with the commit message `Enhanced titles`?

File: sphinx_scality/static/css/current.css_t [hunk 32]
```diff
@@ -314,11 +315,18 @@ input[type=text], select {
 
 /* TITLES */
 
-div.body h1 { margin-top: 0; font-size: 200%; }
-div.body h2 { font-size: 160%; }
-div.body h3 { font-size: 140%; }
-div.body h4 { font-size: 120%; }
-div.body h5 { font-size: 110%; }
+div.body h1 { margin-top: 0;
```
Comment (gdemonet) [inline sphinx_scality/static/css/current.css_t]: Let's try to be consistent with the syntax of CSS rules:

```css
<selector> {
  <attribute>: <value>;
  ...
}
```

File: sphinx_scality/static/css/theme.css [hunk 33]
```diff
@@ -3,6 +3,7 @@
 
 @import url("./fonts.css");
 @import url("./current.css");
+@import url("./mexico.css");
```
Comment (gdemonet) [inline sphinx_scality/static/css/theme.css]: Huh ???

File: sphinx_scality/static/css/theme.css [hunk 34]
```diff
@@ -21,7 +22,7 @@ img.baselogo {
 }
 div.baselogo > a {
   
-    font-size: 23px;
+    font-size: 19px;
```
Comment (gdemonet) [inline sphinx_scality/static/css/theme.css]: Let's try to clean commits please.

File: sphinx_scality/static/css/theme.css [hunk 35]
```diff
@@ -39,8 +40,19 @@ div.logo img {
 
 /*****content******/
 
+a.headerlink {
```
Comment (gdemonet) [inline sphinx_scality/static/css/theme.css]: `headerlink` in the "contents" section? Feels weird.

File: sphinx_scality/static/css/theme.css [hunk 36]
```diff
@@ -39,8 +40,19 @@ div.logo img {
...
+    color: lightslategray;
```
Comment (gdemonet) [inline sphinx_scality/static/css/theme.css]: Theming of colors please!

File: sphinx_scality/static/css/theme.css [hunk 37]
```diff
@@ -39,8 +40,19 @@ div.logo img {
...
+    
+}
+
+
+a.headerlink:hover {
+    background-color: white !important;
```
Comment (gdemonet) [inline sphinx_scality/static/css/theme.css]: Ditto.

File: sphinx_scality/static/css/theme.css [hunk 38]
```diff
@@ -39,8 +40,19 @@ div.logo img {
...
+    color: orange;
+}
+
 div.content.inside {
-    margin: 0% 5% 0 10% !important;
+    margin: 0% 5%% 0 10% !important;
```
Comment (gdemonet) [inline sphinx_scality/static/css/theme.css]: Hum, one `%` extra is not going to make the CSS parser happy

File: sphinx_scality/static/css/current.css_t (hunk 6 above)
Comment (lucieleonard) [reply sphinx_scality/static/css/current.css_t]: This is the solution we found to have "div.fixed-sidebar{
    position: sticky;"
overflow: auto is not compatible with the fixed right menu

File: example/conf.py (hunk 30 above)
Comment (IsabelleTeran) [reply example/conf.py]: miuauqs

File: sphinx_scality/static/js/searchtools.js (hunk 3 above)
Comment (lucieleonard) [reply sphinx_scality/static/js/searchtools.js]: the ${...} do not work (whether $searchterms or $KB_LINK)
Keeping what we already had, but improving the variables (%searchterm instead of %t)
